prost-build/src/code_generator.rs

```diff
@@ -1028,7 +1026,7 @@ impl CodeGenerator<'_> {
             Type::Message => Cow::Borrowed("message"),
             Type::Enum => Cow::Owned(format!(
                 "enumeration={:?}",
-                self.resolve_ident(field.type_name())
+                self.resolve_ident(&FullyQualifiedName::from_type_name(field.type_name()))
```

Comment:
Should `field` return a `FullyQualifiedName` here? That would mean that the new type should be public. Maybe that is too much for this PR.